plutus-tx/src/PlutusTx/Builtins/Internal.hs

```diff
@@ -69,15 +80,16 @@ error = mustBeReplaced "error"
 BOOL
 -}
 
-newtype BuiltinBool = BuiltinBool Bool
+-- See Note [Opaque builtin types]
+data BuiltinBool = BuiltinBool Bool
```

Comment:
I understood that the problem only arose with ByteStrings. Do you err on the side of caution or you have examples that ghc messed with our bools and units as well?